examples/plot_compare_reduction.py

```diff
@@ -72,4 +86,45 @@
 plt.ylabel('Digit classification accuracy')
 plt.ylim((0, 1))
 plt.legend(loc='upper left')
+
+###############################################################################
+# Caching transformers within a ``Pipeline``
```

Comment:
I am trying to fight the proliferation of example (we have too many) and to make features easier to discover.

What we have found in nilearn is that the notebook-style example of sphinx-gallery enable to add sections in examples and hence enhance simple examples without compromising their readability. What's important is that each section contributes one idea and is kept simple.